### PR TITLE
feat: Render row numbers if using rowNumbers

### LINKED ISSUE #1466: [autogrid] support adding "Row Number" column
### Describe your motivation

It would be very nice to have an option for enabling a column showing the row number.

### Describe the solution you'd like

By default it might not be useful to have it enabled. Then it could be enabled by something it like this:

```jsx
<AutoGrid service={...} model={...} rowNumbers />
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Artur-**
Fixes #1466

### CODE REVIEW

File: packages/ts/react-grid/src/autogrid.tsx [hunk 1]
```diff
@@ -136,10 +139,14 @@ function useColumns(
...
     );
   });
+  let columns = autoColumns;
   if (options.customColumns) {
-    return [...autoColumns, ...options.customColumns];
+    columns = [...columns, ...options.customColumns];
   }
-  return autoColumns;
+  if (options.rowNumbers) {
+    columns = [<GridColumn width="4em" renderer={AutoGridRowNumberRenderer}></GridColumn>, ...columns];
```
Comment (sissbruecker) [inline packages/ts/react-grid/src/autogrid.tsx:147]: The column should define a `key`

File: packages/java/tests/spring/react-grid-test/src/test/java/dev/hilla/test/reactgrid/ReadOnlyGridIT.java [hunk 2]
```diff
@@ -5,31 +5,39 @@
...
+        return super.getLastNameColumn() + 1;
+    }
+
     @Test
     public void dataShown() {
         Assert.assertEquals(8, grid.getLastVisibleRowIndex());
+        Assert.assertEquals(1, grid.getCell(0, 0).getText());
         assertName(0, "Abigail", "Carter");
+        Assert.assertEquals(9, grid.getCell(8, 0).getText());
         assertName(8, "Edward", "Gonzalez");
```
Comment (sissbruecker) [inline packages/java/tests/spring/react-grid-test/src/test/java/dev/hilla/test/reactgrid/ReadOnlyGridIT.java:32]:
```suggestion
        Assert.assertEquals(8, grid.getLastVisibleRowIndex());
        Assert.assertEquals("1", grid.getCell(0, 0).getText());
        assertName(0, "Abigail", "Carter");
        Assert.assertEquals("9", grid.getCell(8, 0).getText());
        assertName(8, "Edward", "Gonzalez");
```

File: packages/ts/react-grid/src/autogrid.tsx (hunk 1 above)
Comment (sissbruecker) [reply packages/ts/react-grid/src/autogrid.tsx]: They are wrapped into the column context, which has the key:
https://github.com/vaadin/hilla/blob/3f284b057a1eacaebe87c03a5dd496c8ec997905/packages/ts/react-grid/src/autogrid.tsx#L131-L136

Comment (sissbruecker) [reply packages/ts/react-grid/src/autogrid.tsx]: The delete column in auto crud has the same issue BTW:
https://github.com/vaadin/hilla/blob/3f284b057a1eacaebe87c03a5dd496c8ec997905/packages/ts/react-grid/src/autocrud.tsx#L26-L29

**Review by sissbruecker — APPROVED** [review]

File: packages/ts/react-grid/src/autogrid.tsx (hunk 1 above)
Comment (Artur-) [reply packages/ts/react-grid/src/autogrid.tsx]: Why do the children of a ColumnContext not require a key?

Comment (Artur-) [reply packages/ts/react-grid/src/autogrid.tsx]: Ah, it has only one child. Never mind

**Pull request merged**